File: model/src/main/java/net/artux/pda/model/items/WeaponModel.java
package net.artux.pda.model.items;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class WeaponModel extends WearableModel {

    private float precision;
    private float speed;
    private float damage;
    private float condition;
    private int bulletQuantity;
    private int bulletId;

}

File: model/src/main/java/net/artux/pda/model/items/WeaponModel.kt
package net.artux.pda.model.items

data class WeaponModel(
    var precision: Float = 0f,
    var speed: Float = 0f,
    var damage: Float = 0f,
    var condition: Float = 0f,
    var bulletQuantity: Int = 0,
    var bulletId: Int = 0,
    var sounds: WeaponSound? = null
) : WearableModel() {

    fun getCalcPrecision(): Float{
        if (type === ItemType.PISTOL)
            return precision + 2

        return precision
    }
}